lib/Service/RecipeService.php

```diff
@@ -1130,61 +1098,25 @@ public function parseRecipeFile($file) {
 	 * @return File
 	 */
 	public function getRecipeImageFileByFolderId($id, $size = 'thumb') {
-		if (!$size) {
-			$size = 'thumb';
-		}
-		if ($size !== 'full' && $size !== 'thumb' && $size !== 'thumb16') {
-			throw new Exception('Image size "' . $size . '" not recognised');
+		$recipe_folders = $this->root->getById($id);
+		if (count($recipe_folders) < 1) {
+			throw new Exception($this->il10n->t('Recipe with id %d not found.', [$id]));
```

Comment:
Should we introduce a more specific Exception?

Reply:
We need to break the RecipeService apart into smaller (and testable) pieces. I do not see much benefit in introducing new Exceptions while we do not yet know the final structure. But yes, it is only the agenda. (And yes, I have already a `RecipeNotFoundException` class in my mind).